iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/DataRegion.java

```diff
@@ -3126,7 +3136,9 @@ private void deleteDataInSealedFiles(Collection<TsFileResource> sealedTsFiles, M
       } // else do nothing
     }
 
-    if (!deletedByFiles.isEmpty()) {
+    if (!deletedByFiles.isEmpty()
+        && !isDropMeasurementExist
+        && idPredicateType.equals(IDPredicate.IDPredicateType.NOP)) {
```

Comment:
<img width="1403" height="743" alt="image" src="https://github.com/user-attachments/assets/025a2157-0c31-4a4f-9a0f-ce00a7157ef3" />
The judgement should be put here.